src/averageAges.js

```diff
@@ -55,7 +80,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let children = people.filter(child =>
+    people.some(mother => child.mother === mother.name));
+  const sons = getPeopleBySex(children, 'm');
+
+  if (onlyWithSon) {
+    children = sons;
+  }
+
+  const ageDiff = children.map(child =>
+    child.born - people.find(woman => woman.name === child.mother).born);
```

Comment:
```suggestion
  const ageDiff = children.map(child => (
    child.born - people.find(woman => woman.name === child.mother).born
  ));
```

Reply:
and what if mother wasn't found? it'll look like `undefined.born`, make a variable for mother and then do the calculation with it

Reply:
I'm not sure if I did it right